### PR TITLE
Random Battles: Refactor LGPE and Gens 1-3

### PR DESCRIPTION
**Pull request by AnnikaCodes**
I found no changes in a sample of 10,000 teams/format so this should be fairly regression-free.

@ACakeWearingAHat if you're planning to stay involved with maintaining Gen 2 development, you should probably review this/familiarize yourself with the changes. (You don't have to, though; I don't mind doing all of the programming for Gen 2.)

### CODE REVIEW

File: data/mods/gen1/random-teams.ts [hunk 1]
```diff
@@ -234,6 +230,29 @@ export class RandomGen1Teams extends RandomGen2Teams {
...
+
+		// other redundancies that aren't handled within the movesets themselves
+		case 'selfdestruct':
+			return {cull: hasMove['rest']};
+		case 'rest':
+			return {cull: hasMove['selfdestruct']};
+		case 'sharpen': case 'swordsdance':
+			return {cull: counter.Special > counter.Physical || !counter.Physical || hasMove['growth']};
+		case 'growth':
+			return {cull: counter.Special < counter.Physical || !counter['Special'] || hasMove['swordsdance']};
```
Comment (KrisXV) [inline data/mods/gen1/random-teams.ts:249]: why does this use `counter['Special']` and not `counter.Special`?

File: data/mods/letsgo/random-teams.ts [hunk 2]
```diff
@@ -228,9 +226,14 @@ export class RandomLetsGoTeams extends RandomTeams {
...
 			const species = this.dex.getSpecies(id);
 			if (
-				species.num < 1 || (species.num > 151 && ![808, 809].includes(species.num)) || species.gen > 7 ||
-				species.nfe || !species.randomBattleMoves || !species.randomBattleMoves.length
-			) continue;
+				species.num < 1 ||
+				(species.num > 151 && ![808, 809].includes(species.num)) ||
+				species.gen > 7 ||
+				species.nfe ||
+				!species.randomBattleMoves || !species.randomBattleMoves.length
```
Comment (KrisXV) [inline data/mods/letsgo/random-teams.ts:233]: cant this be optional chained

File: data/mods/letsgo/random-teams.ts [hunk 3]
```diff
@@ -258,14 +261,12 @@ export class RandomLetsGoTeams extends RandomTeams {
 			}
 			if (skip) continue;
 
-			const set = this.randomSet(species, teamDetails);
```
Comment (KrisXV) [inline data/mods/letsgo/random-teams.ts:261]: set was defined independently  for teamDetails counters

**Review by KrisXV — CHANGES_REQUESTED** [review]

File: data/mods/gen1/random-teams.ts (hunk 1 above)
Comment (AnnikaCodes) [reply data/mods/gen1/random-teams.ts:249]: My find and replace regex missed it for some reason.

**Review by KrisXV — APPROVED** [review]

**Pull request merged**